### ISSUE TITLE
Confused by the loos of  line.py

### ISSUE THREAD

**Issue by YuanEZhou** [thunlp/OpenNE#19]
Hi @zzy14 ,thanks for your excellent code. I am confused by the loos function of line.py, specifically,
1.The commented code of line.py (43-48) is the original form of the paper, it is easy to understand. I want to know whether some minus sign are missed here. 
`# self.sample_sum2 =**-** tf.reduce_sum(tf.log(tf.nn.sigmoid(**-**tf.reduce_sum(tf.multiply(self.pos_h_v_e, self.neg_t_e_context), axis=2))), axis=1)
        # self.second_loss = tf.reduce_mean(-tf.log(tf.nn.sigmoid(tf.reduce_sum(tf.multiply(self.pos_h_e, self.pos_t_e_context), axis=1))) +
        #                            self.sample_sum2)
        # self.sample_sum1 = **-**tf.reduce_sum(tf.log(tf.nn.sigmoid(**-**tf.reduce_sum(tf.multiply(self.pos_h_v_e, self.neg_t_e), axis=2))), axis=1)
        # self.first_loss = tf.reduce_mean(-tf.log(tf.nn.sigmoid(tf.reduce_sum(tf.multiply(self.pos_h_e, self.pos_t_e), axis=1))) +
        #                            self.sample_sum1)`
2. The loos funcution you actually used in the code (49-54) is the reduced form of original paper?

**Comment by zzy14** [comment]
Hi @YuanEZhou , thanks for your question. In the commented code, some minus are in the wrong place. I will fix it soon. And the reason why I used a different form is that I found the original loss can not drop in mini-batch training. This week, I found the negative sampling with mini-batch prevents the original loss from dropping. So I will push a new version of LINE recently. It will use the original loss.
